### PR TITLE
Improve slap for import command

### LINKED ISSUE #164: Improve slap
Improve slap in my execute function

_Originally posted by @jayllo-c in https://github.com/AY2324S2-CS2103T-T10-1/tp/pull/162#discussion_r1532142517_

### PR DESCRIPTION
**Pull request by Pughal77**
Improve SLAP for import command

Let's work towards better code readability

fixes #164

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ImportCommand.java [hunk 1]
```diff
@@ -109,21 +119,29 @@ public List<Map<String, String>> readCsvFile() throws DataLoadingException {
...
     }
+    /**
+     * Parses the data from the csv file into a list of maps. Each map represents a person's data.
+     * @param rows
+     * @return
+     */
+    public List<Map<String, String>> parseData(List<String[]> rows) {
+        List<Map<String, String>> data = new ArrayList<>();
+        String[] header = rows.get(0);
+        for (int i = 1; i < rows.size(); i++) {
```
Comment (Zer0Legion) [inline src/main/java/seedu/address/logic/commands/ImportCommand.java:135]: is there a way to remove this nested for loop?

File: src/test/data/ExportCommandTest/addressbookdata/addressbooktest.csv
Comment (Zer0Legion) [inline src/test/data/ExportCommandTest/addressbookdata/addressbooktest.csv:1]: i dont think this file should be merged in

File: src/test/data/ExportCommandTest/filteredAddressBook.json
Comment (Zer0Legion) [inline src/test/data/ExportCommandTest/filteredAddressBook.json:1]: i dont think this file should be merged in

**Review by Zer0Legion — COMMENTED** [review]
Overall great SLAP principles are used here! Huge upgrade from the previous iteration

File: docs/DeveloperGuide.md
Comment (jayllo-c) [inline docs/DeveloperGuide.md:1]: Might wanna change the PR title to reflect the changes you've made to the DG

File: src/main/java/seedu/address/logic/parser/ImportCommandParser.java
Comment (jayllo-c) [inline src/main/java/seedu/address/logic/parser/ImportCommandParser.java:1]: Appreciate the further clarification on the path here!

**Review by jayllo-c — COMMENTED** [review]
Good improvement on the SLAP. Just take note of the comments.

**Review by delishad21 — COMMENTED** [review]
Can see that slap is indeed being used. Your DG changes seem to appear here as well. Is it meant to be in this PR? or have you branched incorrectly?

**Pull request closed without merging**